sprokit/processes/vxl/kw_archive_writer_process.cxx

```diff
@@ -267,19 +293,75 @@ ::_step()
   kwiver::vital::timestamp frame_time = grab_input_using_trait( timestamp );
 
   // image
-  //+ kwiver::vital::image_container_sptr img = grab_input_as< kwiver::vital::image_container_sptr > ( priv::port_image );
   kwiver::vital::image_container_sptr img = grab_from_port_using_trait( image );
+
+  // check for empty image
+  if( !img || img->width() == 0 || img->height() == 0 )
+  {
+    push_to_port_using_trait( complete_flag, true );
+    return;
+  }
+
   kwiver::vital::image image = img->get_image();
 
   // homography
-  //+ kwiver::f2f_homography homog = grab_input_as< kwiver::vital::f2f_homography > ( priv::port_src_to_ref_homography );
-  kwiver::vital::f2f_homography homog = grab_from_port_using_trait( homography_src_to_ref );
+  kwiver::vital::f2f_homography homog( Eigen::Matrix< double, 3, 3 >(), -1, -1 );
+
+  if( process::has_input_port_edge( "homography_src_to_ref" ) )
+  {
+    homog = grab_from_port_using_trait( homography_src_to_ref );
+  }
 
   // corners
-  kwiver::vital::geo_polygon corners = grab_input_using_trait( corner_points );
+  kwiver::vital::geo_polygon corners;
+
+  if( process::has_input_port_edge( "corner_points" ) )
+  {
+    corners = grab_input_using_trait( corner_points );
+  }
```

Comment:
Removing these checks causes the program to either crash or hang forever

Reply:
2018-11-02 18:00:42.895 DEBUG video_input_image_list.cxx(385): reading image from file "../../example_imagery/small_example_image_set1/03149_D20160630-T162921.356_00-0C-DF-06-40-BF.png"
2018-11-02 18:00:42.895 DEBUG image_io.cxx(346): Loading image from file: ../../example_imagery/small_example_image_set1/03149_D20160630-T162921.356_00-0C-DF-06-40-BF.png
Exception in thread kwa_writer:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/matt/Dev/viame/build/install/lib/python2.7/dist-packages/sprokit/schedulers/pythread_per_process.py", line 116, in _run_process
    proc.step()
RuntimeError: failed to convert from string representation "0 0 0 0 0 0 0 0", thrown from /home/matt/Dev/viame/src/packages/kwiver/vital/types/geo_polygon.cxx:139

2018-11-02 18:03:25.946 DEBUG image_io.cxx(346): Loading image from file: ../../example_imagery/small_example_image_set1/03149_D20160630-T162921.356_00-0C-DF-06-40-BF.png
Exception in thread kwa_writer:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/matt/Dev/viame/build/install/lib/python2.7/dist-packages/sprokit/schedulers/pythread_per_process.py", line 116, in _run_process
    proc.step()
RuntimeError: The port 'homography_src_to_ref' on process 'kwa_writer' is not connected: Data was requested from the port, thrown from /home/matt/Dev/viame/src/packages/kwiver/sprokit/src/sprokit/pipeline/process.cxx:1446

2018-11-02 18:06:33.300 DEBUG image_io.cxx(346): Loading image from file: ../../example_imagery/small_example_image_set1/03149_D20160630-T162921.356_00-0C-DF-06-40-BF.png
Exception in thread kwa_writer:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/matt/Dev/viame/build/install/lib/python2.7/dist-packages/sprokit/schedulers/pythread_per_process.py", line 116, in _run_process
    proc.step()
RuntimeError: failed to convert from string representation "common:fixed_gsd", thrown from /home/matt/Dev/viame/src/packages/kwiver/vital/config/config_block.h:547, thrown from /home/matt/Dev/viame/src/packages/kwiver/vital/config/config_block.h:554

Particularly for the  3 parameters, the homography because it's not static - but the others because the defaults are wrong or because they're not specified in the pipe. They don't need to be specified in the .pipe, they are optional parameters. GSD doesn't need to be static, it can change in the pipeline sense.

Reply:
For now I just removed the static option, but if the 'if' checks could be changed to if ( process::has_input_port_edge( "corner_points" ) || process::has_input_port_static( "corner_points" ) ) or something that might work as well? Assuming that 2nd function existed